internal/fileutils/fileutils.go

```diff
@@ -681,6 +696,56 @@ func WriteTempFileToDir(dir, pattern string, data []byte, perm os.FileMode) (str
 	return f.Name(), nil
 }
 
+func CopyFilesFromAssets(assetDir, dst string) error {
+	entries, err := assets.ReadDir(assetDir)
+	if err != nil {
+		return errs.Wrap(err, "asset.ReadDir %s failed", assetDir)
+	}
+
+	for _, entry := range entries {
+		srcPath := filepath.Join(assetDir, entry.Name())
+		destPath := filepath.Join(dst, entry.Name())
+
+		switch entry.Type() & os.ModeType {
+		case os.ModeDir:
+			err := MkdirUnlessExists(destPath)
+			if err != nil {
+				return errs.Wrap(err, "MkdirUnlessExists %s failed", destPath)
+			}
+
+			err = CopyFilesFromAssets(srcPath, destPath)
+			if err != nil {
+				return errs.Wrap(err, "CopyFiles %s:%s failed", srcPath, destPath)
+			}
+		default:
+			err := CopyAsset(srcPath, destPath)
+			if err != nil {
+				return errs.Wrap(err, "CopyFile %s:%s failed", srcPath, destPath)
+			}
+		}
+	}
+
+	walkErr := filepath.Walk(dst, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.Name() == assets.PlaceholderFileName {
+			err := os.Remove(path)
+			if err != nil {
+				return errs.Wrap(err, "os.Remove %s failed", path)
+			}
+		}
+
+		return nil
+	})
+	if walkErr != nil {
+		return errs.Wrap(walkErr, "filepath.Walk %s failed", dst)
+	}
```

Comment:
This feels pointless. We are copying all files and then walking over the files we copied to remove these.. why not skip copying these in the first place (line ~721)?

Reply:
🤦‍♂️